vpr/src/base/SetupVPR.cpp

```diff
@@ -97,6 +99,7 @@ void SetupVPR(const t_options* Options,
     FileNameOpts->CmosTechFile = Options->CmosTechFile;
     FileNameOpts->out_file_prefix = Options->out_file_prefix;
     FileNameOpts->read_vpr_constraints_file = Options->read_vpr_constraints_file;
+    FileNameOpts->read_xdc_constraints_file = Options->XDCFile;
```

Comment:
I'd avoid having an option that is strictly related to XDC, or any other constraint file type.

I think it may be better to use the already present `read_vpr_constraints_file`, and maybe rename it to `read_constraints_file` and, based on the extension of this file, choose the right way to parse it.

Reply:
Hmm, I'm not sure about it.
Currently we need BOTH SDC and XDC files, because a separate parser for SDC exists. Ideally both XDC and SDC should use the same backend for reading the files. I mean, we could maybe allow passing both of these files with a single flag, but there would be a hard limit of two files of different format, something I've never seen being done with a CLI interface. Alternatively we can allow reading more XDC files and thus make the flag accept any number of paths. Vivado (and F4PGA) allow reading multiple XDCs, so...

Reply:
The problem here is that we'd need to add one option for each kind of different constraint files that we provide. The constraint that we are considering here though is only a placement constraint kind of file. We do not care about the SDC at the moment, as it is parsed differently and does not concern constraining blocks to specific locations.

Just imagine that we'll add support for PCF and other kinds of constraint files. We'd need to invoke VPR with a different option everytime. Having instead a dynamic way of determining what is the type of constraint (for now based on the extension of the constraint file), allows us to avoid having multiple options that in the end are doing the same thing, e.g. telling VPR that there is a constraint file to parse.

Reply:
For now I'm leaving it at having separate argument. We don't infer architecture format from file extension, so why should we do that for constraints. Additionally, I made it possible to pass multiple XDC constraint files (the flag is now called `--xdc_files`).